cmd/restic/cmd_backup.go

```diff
@@ -645,6 +669,8 @@ func runBackup(ctx context.Context, opts BackupOptions, gopts GlobalOptions, ter
 		Time:           timeStamp,
 		Hostname:       opts.Host,
 		ParentSnapshot: parentSnapshot,
+		Command:        command,
+		CommandStderr:  stderr,
```

Comment:
There's no need to modify the archiver. Instead, the `Close() error` part of the `ReadCloser` of `fs.Reader` should wait for the command to exit and return an error if necessary.